src/Microsoft.Sbom.Api/Executors/LicenseInformationService.cs

```diff
@@ -0,0 +1,85 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading.Tasks;
+using Serilog;
+
+namespace Microsoft.Sbom.Api.Executors;
+public class LicenseInformationService
+{
+    private readonly ILogger log;
+
+    public LicenseInformationService(ILogger log)
+    {
+        this.log = log ?? throw new ArgumentNullException(nameof(log));
+    }
+
+    public async Task<List<string>> FetchLicenseInformationFromAPI(List<string> listOfComponentsForApi)
+    {
+        int batchSize = 400;
+        List<HttpResponseMessage> responses = new List<HttpResponseMessage>();
+        List<string> responseContent = new List<string>();
+
+        for (int i = 0; i < listOfComponentsForApi.Count; i += batchSize)
+        {
+            List<string> batch = listOfComponentsForApi.Skip(i).Take(batchSize).ToList();
+            string formattedData = "[" + string.Join(",", batch.Select(item => $"\"{item}\"")) + "]";
+
+            log.Debug($"Retrieving license information for {batch.Count} components...");
+
+            using (HttpClient httpClient = new HttpClient())
```

Comment:
Don't `new` this up. Accept it as a constructor parameter. See https://learn.microsoft.com/en-us/aspnet/core/fundamentals/http-requests?view=aspnetcore-7.0#typed-clients

Reply:
This one I wasn't too sure about. Would I have to move over some things to a TypedClientModel class as shown in the docs? And I couldn't figure out where the HttpClient would actually get instantiated I was able to test it out a bit and doing a .AddHTTPClient<LicenseInformationService>() in the program.cs works but I'm not sure how adding this would affect people using our API directly.